### DISCUSSION TITLE
How to use grain wasm files in Deno

### DISCUSSION THREAD

**Discussion by Mishieck** [grain-lang/grain#1773] [Help]
Hi,

I would like to write code using Grain, convert the code to WebAssembly, and finally use the WebAssembly code in Deno.

Let's say I have a Grain module `is-even.gr` with the following code:

```js
let isEven = (value) => value % 2 == 0;

export isEven;
```
I would like to be able to transpile the code to a wasm file that is compatible with Deno. I would like to use the wasm code as demonstrated in the Deno script below.

```ts
const wasmCode = await Deno.readFile('is-even.gr.wasm');
const wasmModule = new WebAssembly.Module(wasmCode);
const wasmInstance = new WebAssembly.Instance(wasmModule, {});
const isEven = wasmInstance.exports.isEven as CallableFunction;
console.log(isEven(2));
console.log(isEven(1));
```
I've tried to use the wasm file generated using `grain is-even.gr`, but it doesn't work. I'm getting the following error:

```bash
error: Uncaught (in promise) TypeError: WebAssembly.Instance(): Import #0 module="GRAIN$MODULE$runtime/gc" error: module is not an object or function
const wasmInstance = new WebAssembly.Instance(wasmModule, {});
```

**Comment by phated** [comment]
I'll transfer this to a discussion.

**Comment by Mishieck** [comment]
Ok. Thanks.

**Comment by spotandjake** [comment]
https://deno.land/manual@v1.8.3/getting_started/webassembly there is the docs with using Wasm with Deno, you would basically just take your `<compiledFileName>.gr.wasm` and use the docs there to run the contents inside. to get the compiled file you would run`grain compiler <compiledFileName>.gr --release`.

You might need to use the `wasi_polyfill` as deno doesn't have built in support for wasi, something like wasmers bindings might be an option as well.
